ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveOptimizeInlineArrayTableFunctionRule.java

```diff
@@ -124,9 +125,12 @@ public void onMatch(RelOptRuleCall call) {
     RexNode newInlineCall =
         cluster.getRexBuilder().makeCall(tfs.getRowType(), inlineCall.op, newArrayCall);
 
+    // Use empty listfor columnMappings. The return row type of the RelNode now comprises of
+    // all the fields within the UDTF, so there is no mapping from the output fields
+    // directly to the input fields anymore.
     final RelNode newTableFunctionScanNode = tfs.copy(tfs.getTraitSet(),
         tfs.getInputs(), newInlineCall, tfs.getElementType(), tfs.getRowType(),
-        tfs.getColumnMappings());
+        Collections.emptySet());
 
     call.transformTo(newTableFunctionScanNode);
```

Comment:
Does this rule still work as it is now that we are wrapping the table function with the `LATERAL` operator? Do we want/need to add special cases if we have lateral + inline array?

Reply:
In my writeup in an earlier comment, I mentioned this rule as being extraneous if we go with option 1) and cause the regression with sql statements which contain  lateral view inline(array(...)).

Reply:
I saw the new changes in `LateralViewPlan` and this made me thinking that an equivalent thing could be done as part of this rule. Something like match `lateral(inline(array(1,2)))` and transform to `inline(array($0,1,2))`.

I don't feel strongly about it so if you prefer to keep the `LateralViewPlan.isInlineArray` method there I am OK with it.